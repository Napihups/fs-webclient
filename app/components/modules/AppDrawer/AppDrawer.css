.appDrawer {
  @apply fixed 
  top-0 
  h-screen 
  bg-base-100 
  flex 
  flex-col 
  justify-between
  transition
  border-r-[1px] border-base-200;
}

.appDrawer__logobox {
  @apply flex 
  flex-row 
  items-center 
  justify-center
  h-[70px];
}
.appDrawer__logo {
  @apply scale-50;
}

.appDrawer__menu {
  @apply list-none
  flex flex-col
  gap-2
  px-[10px]
  m-0 overflow-scroll;
}
.appDrawer__menu li {
  @apply btn
  btn-outline
  text-base-content
  hover:bg-primary
  hover:border-primary
  hover:text-primary-content
  pl-[20px] 
  min-w-[200px]
  border-transparent 
  flex 
  flex-row 
  items-center 
  justify-start 
  normal-case
  font-normal;
}
.appDrawer__menu li.active {
  @apply bg-primary border-primary text-primary-content;
}

.appDrawer__menuText {
  @apply ml-3;
}

.appDrawer__toggler {
  @apply btn text-base-content  bg-transparent border-transparent hover:bg-transparent hover:border-transparent mx-[20px] mb-[20px];
}
.appDrawer__themeSwitch {
  @apply btn  text-base-content bg-transparent border-transparent hover:bg-transparent hover:border-transparent mx-[20px] mb-[20px];
}
