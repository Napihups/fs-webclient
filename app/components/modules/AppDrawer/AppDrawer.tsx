import React, { useState } from "react";
import { motion, AnimatePresence } from "framer-motion";
import Link from "next/link";
import { LogoDark } from "@element/Logo/LogoDark";
import { LogoLight } from "@element/Logo/LogoLight";
import { LogoGeometric } from "@element/Logo/LogoGeometric";
import { DrawerNavigationMenuItem, DrawerNavigationMenus } from "@constant/drawer-navigation-menu";
import { useAppTheme } from "@common/AppTheme";
import { ThemeLightIcon } from "@element/Icons/ThemeLightIcon";
import { ThemeDarkIcon } from "@element/Icons/ThemeDarkIcon";
import { DrawerCollapseIcon } from "@element/Icons/DrawerCollapseIcon";
import { DrawerExpandIcon } from "@element/Icons/DrawerExpandIcon";

const DRAWER_EXPAND_WIDTH = "260px";
const DRAWER_COLLAPSE_WIDTH = "85px";

export const AppDrawer: React.FC = () => {
  const [collapse, setCollapse] = useState<boolean>(false);

  const { theme, setTheme } = useAppTheme();

  const onToggle = (ev: React.ChangeEvent<HTMLInputElement>) => {
    setCollapse(!ev.target.checked);
  };

  const onChangeTheme = (ev: React.ChangeEvent<HTMLInputElement>) => {
    const checked = ev.target.checked;

    setTheme(checked ? "dark" : "light");
  };

  return (
    <AnimatePresence initial={false}>
      <motion.div
        initial={{ width: collapse ? DRAWER_EXPAND_WIDTH : DRAWER_COLLAPSE_WIDTH }}
        animate={{ width: collapse ? DRAWER_COLLAPSE_WIDTH : DRAWER_EXPAND_WIDTH }}
        transition={{ delay: collapse ? 0.15 : 0 }}
        className="appDrawer"
        role="navigation"
      >
        <div className="w-full">
          <div className="appDrawer__logobox">
            {!collapse && (
              <motion.div
                initial={{ opacity: 0 }}
                animate={{ opacity: 1 }}
                exit={{ opacity: 0 }}
                className="appDrawer__logo"
              >
                {theme === "dark" ? <LogoDark /> : <LogoLight />}
              </motion.div>
            )}
            {collapse && (
              <motion.div
                initial={{ opacity: 0, rotate: "180deg", scale: "50%" }}
                animate={{ opacity: 1, rotate: "360deg", scale: "50%" }}
                transition={{
                  rotate: {
                    delay: 0.2,
                  },
                }}
                className="appDrawer__logo"
              >
                <LogoGeometric />
              </motion.div>
            )}
          </div>

          <motion.ul role={"menu"} className="appDrawer__menu">
            {DrawerNavigationMenus.map((item: DrawerNavigationMenuItem) => {
              const { icon: Icon, label, href, index } = item;
              return (
                <Link key={index} href={href}>
                  <li role={"menuitem"}>
                    <Icon />
                    <AnimatePresence>
                      {!collapse && (
                        <motion.span
                          initial={{ opacity: 0, x: 0 }}
                          animate={{ opacity: 1, x: 0 }}
                          exit={{ opacity: 0, x: 0 }}
                          className="appDrawer__menuText"
                        >
                          {label}
                        </motion.span>
                      )}
                    </AnimatePresence>{" "}
                  </li>
                </Link>
              );
            })}
          </motion.ul>
        </div>

        <div className="flex flex-col">
          <label className="appDrawer__themeSwitch swap swap-rotate">
            <input type="checkbox" onChange={onChangeTheme} />
            <ThemeDarkIcon className="swap-off fill-current w-8 h-8" />
            <ThemeLightIcon className="swap-on fill-current w-8 h-8" />
          </label>

          <label className="appDrawer__toggler swap swap-rotate">
            <input
              type="checkbox"
              defaultChecked
              data-testid="drawer-toggler"
              onChange={onToggle}
            />
            <DrawerExpandIcon className="swap-off fill-current" />
            <DrawerCollapseIcon className="swap-on fill-current" />
          </label>
        </div>
      </motion.div>
    </AnimatePresence>
  );
};
