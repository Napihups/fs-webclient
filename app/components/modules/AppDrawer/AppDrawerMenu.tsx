import { DrawerNavigationMenus, DrawerNavigationMenuItem } from "@constant/drawer-navigation-menu";
import { motion, AnimatePresence } from "framer-motion";
import Link from "next/link";
import { useRouter } from "next/router";
import React from "react";

const _setMenuItemActiveClass = (currentPath: string, itemPath: string) => {
  return currentPath === itemPath ? "active" : "";
};

type AppDrawerMenuProps = {
  collapse: boolean;
};
export const AppDrawerMenu: React.FC<AppDrawerMenuProps> = ({ collapse }) => {
  const router = useRouter();

  return (
    <motion.ul role={"menu"} className="appDrawer__menu">
      {DrawerNavigationMenus.map((item: DrawerNavigationMenuItem) => {
        const { icon: Icon, label, href, index } = item;
        return (
          <Link key={index} href={href}>
            <li role={"menuitem"} className={`${_setMenuItemActiveClass(router.pathname, href)}`}>
              <Icon />
              <AnimatePresence initial={false}>
                {!collapse && (
                  <motion.span
                    initial={{ opacity: 0, x: 0 }}
                    animate={{ opacity: 1, x: 0 }}
                    exit={{ opacity: 0, x: 0 }}
                    className="appDrawer__menuText"
                  >
                    {label}
                  </motion.span>
                )}
              </AnimatePresence>{" "}
            </li>
          </Link>
        );
      })}
    </motion.ul>
  );
};
