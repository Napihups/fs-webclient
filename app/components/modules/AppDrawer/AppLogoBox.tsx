import { useAppTheme } from "@common/AppTheme";
import { LogoDark } from "@element/Logo/LogoDark";
import { LogoGeometric } from "@element/Logo/LogoGeometric";
import { LogoLight } from "@element/Logo/LogoLight";
import { motion } from "framer-motion";
import React from "react";

type AppLogoBoxProps = {
  collapse: boolean;
};
export const AppLogoBox: React.FC<AppLogoBoxProps> = ({ collapse }) => {
  const { theme } = useAppTheme();
  return (
    <div className="appDrawer__logobox">
      {!collapse && (
        <motion.div
          initial={{ opacity: 0 }}
          animate={{ opacity: 1 }}
          exit={{ opacity: 0 }}
          className="appDrawer__logo"
        >
          {theme === "fsDark" ? <LogoDark /> : <LogoLight />}
        </motion.div>
      )}
      {collapse && (
        <motion.div
          initial={{ opacity: 0, rotate: "180deg", scale: "50%" }}
          animate={{ opacity: 1, rotate: "360deg", scale: "50%" }}
          transition={{
            rotate: {
              delay: 0.2,
            },
          }}
          className="appDrawer__logo"
        >
          <LogoGeometric />
        </motion.div>
      )}
    </div>
  );
};
