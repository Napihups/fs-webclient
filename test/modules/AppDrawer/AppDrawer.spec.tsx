import { screen, render, cleanup, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { AppDrawer } from "@module/AppDrawer/AppDrawer";

describe("AppDrawer UI test", () => {
  afterEach(() => {
    cleanup();
  });
  test("Static: Textual for all element in the AppDrawer should be as per requirement", async () => {
    const staticTextItems = {
      home: "Home",
      finance: "Finance",
      calendar: "Calendar",
      timelines: "Timeline",
      diary: "Diary",
      setting: "Settings",
    };

    render(<AppDrawer />);

    const drawer = await screen.findByRole("navigation");

    await screen.findByText(staticTextItems.home);
    await screen.findByText(staticTextItems.finance);
    await screen.findByText(staticTextItems.calendar);
    await screen.findByText(staticTextItems.timelines);
    await screen.findByText(staticTextItems.diary);
    await screen.findByText(staticTextItems.setting);

    expect(drawer).toMatchSnapshot();
  });

  test("Static: Default Drawer should have a width of 260px", async () => {
    render(<AppDrawer />);

    const drawer = screen.getByRole("navigation");
    expect(drawer.style.width).toBe("260px");
  });
  test("Testcase: The logo display on the drawer should change when user toggling", async () => {
    const user = userEvent.setup();
    render(<AppDrawer />);
    const toggler = await screen.findByTestId("drawer-toggler");
    expect(await screen.getByTestId("LogoDark")).toBeDefined();
    await user.click(toggler);

    await new Promise(process.nextTick);

    expect(await screen.findByTestId("LogoGeometric")).toBeDefined();
  });
});
