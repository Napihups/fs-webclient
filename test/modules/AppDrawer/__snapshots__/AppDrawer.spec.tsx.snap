// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDrawer UI test Static: Textual for all element in the AppDrawer should be as per requirement 1`] = `
<div
  class="appDrawer"
  role="navigation"
  style="width: 260px;"
>
  <div
    class="w-full"
  >
    <div
      class="appDrawer__logobox"
    >
      <div
        class="appDrawer__logo"
        style="opacity: 1;"
      >
        <svg
          data-testid="LogoDark"
          fill="none"
          height="60"
          width="229"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M29.554 50.927a.968.968 0 0 0 1.229.654c4.968-1.659 9.229-4.918 12.076-9.26a22.425 22.425 0 0 0 3.453-15.573c-.8-5.43-3.569-10.394-7.802-13.984S28.865 7.198 23.266 7.2c-5.6.002-11.01 1.982-15.24 5.575-4.23 3.594-6.996 8.561-7.79 13.99A22.425 22.425 0 0 0 3.7 42.335c2.85 4.34 7.114 7.597 12.083 9.252a.968.968 0 0 0 1.228-.655l2.57-8.59c.16-.533-.149-1.09-.67-1.287a12.051 12.051 0 0 1-5.813-4.642 11.657 11.657 0 0 1-1.801-8.094 11.787 11.787 0 0 1 4.05-7.273 12.248 12.248 0 0 1 7.922-2.898 12.248 12.248 0 0 1 7.925 2.892 11.788 11.788 0 0 1 4.055 7.27 11.658 11.658 0 0 1-1.795 8.095 12.05 12.05 0 0 1-5.809 4.647c-.52.197-.829.755-.669 1.288l2.577 8.587Z"
            fill="url(#a)"
          />
          <path
            d="M75.18 21.205c3.03 0 5.46.9 7.29 2.7 1.86 1.8 2.79 4.515 2.79 8.145V46h-8.55V33.445c0-3.33-1.335-4.995-4.005-4.995-1.47 0-2.655.48-3.555 1.44-.87.96-1.305 2.4-1.305 4.32V46h-8.55V21.61h8.145v2.655a9.44 9.44 0 0 1 3.42-2.25c1.32-.54 2.76-.81 4.32-.81Zm41.584.405V46h-8.145v-2.43c-1.59 1.89-3.99 2.835-7.2 2.835-2.22 0-4.245-.51-6.075-1.53-1.8-1.05-3.24-2.535-4.32-4.455-1.05-1.92-1.575-4.14-1.575-6.66 0-2.52.525-4.725 1.575-6.615 1.08-1.92 2.52-3.39 4.32-4.41 1.83-1.02 3.855-1.53 6.075-1.53 2.88 0 5.145.84 6.795 2.52V21.61h8.55Zm-13.545 18c1.5 0 2.73-.51 3.69-1.53.96-1.05 1.44-2.49 1.44-4.32 0-1.8-.48-3.21-1.44-4.23-.96-1.02-2.19-1.53-3.69-1.53s-2.73.51-3.69 1.53-1.44 2.43-1.44 4.23c0 1.83.48 3.27 1.44 4.32.96 1.02 2.19 1.53 3.69 1.53Zm34.394-18.405c2.22 0 4.23.525 6.03 1.575 1.83 1.02 3.27 2.49 4.32 4.41 1.08 1.89 1.62 4.095 1.62 6.615 0 2.52-.54 4.74-1.62 6.66-1.05 1.89-2.49 3.36-4.32 4.41-1.8 1.02-3.81 1.53-6.03 1.53-2.85 0-5.115-.855-6.795-2.565v10.89h-8.55V21.61h8.145v2.43c1.59-1.89 3.99-2.835 7.2-2.835Zm-1.8 18.405c1.5 0 2.73-.51 3.69-1.53.96-1.05 1.44-2.475 1.44-4.275 0-1.8-.48-3.21-1.44-4.23-.96-1.05-2.19-1.575-3.69-1.575s-2.73.525-3.69 1.575c-.96 1.02-1.44 2.43-1.44 4.23 0 1.8.48 3.225 1.44 4.275.96 1.02 2.19 1.53 3.69 1.53Zm17.744-18h8.55V46h-8.55V21.61Zm4.275-2.7c-1.56 0-2.82-.435-3.78-1.305-.96-.87-1.44-1.95-1.44-3.24 0-1.29.48-2.37 1.44-3.24.96-.87 2.22-1.305 3.78-1.305 1.56 0 2.82.42 3.78 1.26.96.81 1.44 1.86 1.44 3.15 0 1.35-.48 2.475-1.44 3.375-.96.87-2.22 1.305-3.78 1.305Zm36.067 2.7v19.935c0 4.53-1.23 7.92-3.69 10.17-2.46 2.28-6 3.42-10.62 3.42-2.4 0-4.65-.27-6.75-.81-2.1-.54-3.885-1.335-5.355-2.385l3.105-5.985c.99.81 2.235 1.44 3.735 1.89 1.53.48 3.03.72 4.5.72 2.25 0 3.9-.495 4.95-1.485 1.05-.99 1.575-2.46 1.575-4.41v-.81c-1.65 1.92-4.065 2.88-7.245 2.88-2.16 0-4.155-.48-5.985-1.44a11.352 11.352 0 0 1-4.365-4.14c-1.08-1.8-1.62-3.87-1.62-6.21 0-2.31.54-4.35 1.62-6.12 1.08-1.8 2.535-3.18 4.365-4.14 1.83-.99 3.825-1.485 5.985-1.485 3.48 0 6.03 1.125 7.65 3.375v-2.97h8.145Zm-13.77 16.335c1.53 0 2.79-.465 3.78-1.395 1.02-.93 1.53-2.13 1.53-3.6 0-1.47-.495-2.655-1.485-3.555-.99-.93-2.265-1.395-3.825-1.395s-2.85.465-3.87 1.395c-.99.9-1.485 2.085-1.485 3.555 0 1.47.51 2.67 1.53 3.6 1.02.93 2.295 1.395 3.825 1.395Zm31.503 8.46c-2.61 0-4.965-.54-7.065-1.62-2.1-1.08-3.75-2.58-4.95-4.5-1.17-1.92-1.755-4.095-1.755-6.525 0-2.4.585-4.56 1.755-6.48 1.2-1.92 2.835-3.405 4.905-4.455 2.1-1.08 4.47-1.62 7.11-1.62 2.64 0 5.01.54 7.11 1.62 2.1 1.05 3.735 2.535 4.905 4.455 1.17 1.89 1.755 4.05 1.755 6.48 0 2.43-.585 4.605-1.755 6.525s-2.805 3.42-4.905 4.5-4.47 1.62-7.11 1.62Zm0-6.795c1.5 0 2.73-.51 3.69-1.53.96-1.05 1.44-2.49 1.44-4.32 0-1.8-.48-3.21-1.44-4.23-.96-1.02-2.19-1.53-3.69-1.53s-2.73.51-3.69 1.53-1.44 2.43-1.44 4.23c0 1.83.48 3.27 1.44 4.32.96 1.02 2.19 1.53 3.69 1.53Z"
            fill="#fff"
          />
          <defs>
            <lineargradient
              gradientUnits="userSpaceOnUse"
              id="a"
              x1="46.55"
              x2="-25.341"
              y1="52.8"
              y2="-0.982"
            >
              <stop
                stop-color="#03705C"
              />
              <stop
                offset="1"
                stop-color="#5AB8A7"
              />
            </lineargradient>
          </defs>
        </svg>
      </div>
    </div>
    <ul
      class="appDrawer__menu"
      role="menu"
    >
      <li
        class="active"
        role="menuitem"
      >
        <svg
          fill="none"
          height="26"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.143 25.5H5.86A5.867 5.867 0 0 1 0 19.64v-8.218c0-1.816.862-3.558 2.306-4.66l6.642-5.065a5.874 5.874 0 0 1 7.108 0l2.11 1.609V2.059c.052-1.296 1.903-1.295 1.954 0v3.219a.977.977 0 0 1-1.569.777l-3.68-2.805a3.916 3.916 0 0 0-4.739 0L3.492 8.316a3.93 3.93 0 0 0-1.538 3.106v8.218a3.911 3.911 0 0 0 3.907 3.907h13.283a3.911 3.911 0 0 0 3.907-3.907v-8.218c0-1.226-.568-2.39-1.52-3.117a.977.977 0 1 1 1.185-1.553c1.433 1.093 2.289 2.839 2.289 4.67v8.218a5.867 5.867 0 0 1-5.86 5.86Zm-8.595-13.723a1.22 1.22 0 0 0-1.22 1.221c.064 1.62 2.377 1.619 2.441 0a1.22 1.22 0 0 0-1.22-1.22Zm5.128 1.221c-.064 1.62-2.378 1.619-2.442 0 .065-1.62 2.378-1.618 2.442 0Zm-3.907 3.907c-.064 1.62-2.378 1.619-2.441 0 .064-1.62 2.377-1.619 2.441 0Zm3.907 0c-.064 1.62-2.378 1.619-2.442 0 .065-1.62 2.378-1.619 2.442 0Z"
            fill="currentColor"
          />
        </svg>
        <span
          class="appDrawer__menuText"
          style="opacity: 1; transform: none;"
        >
          Home
        </span>
         
      </li>
      <li
        class=""
        role="menuitem"
      >
        <svg
          fill="none"
          height="26"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M22.398 2.952 14.706.806a8.299 8.299 0 0 0-4.471.002L2.588 2.951c-3.937 1.205-3.164 6.877.958 6.973h17.898c4.125-.097 4.893-5.772.954-6.972Zm-.954 5.019H3.546c-1.855-.043-2.203-2.598-.431-3.14l7.647-2.142a6.347 6.347 0 0 1 3.419-.002l7.692 2.146c1.774.541 1.426 3.095-.43 3.138Zm-6.996 13.623a.977.977 0 0 1-.977-.977v-7.861c.052-1.296 1.902-1.295 1.954 0v7.861c0 .54-.438.977-.977.977Zm-3.955 0a.977.977 0 0 1-.977-.977v-7.861c.052-1.296 1.902-1.295 1.953 0v7.861c0 .54-.437.977-.976.977Zm-7.813.049a.977.977 0 0 1-.976-.977v-7.91c.051-1.296 1.902-1.295 1.953 0v7.91c0 .54-.437.977-.977.977Zm3.955-.05a.977.977 0 0 1-.976-.976v-7.861c.051-1.296 1.902-1.295 1.953 0v7.861c0 .54-.437.977-.977.977Zm10.059 2.93c0 .54-.437.977-.976.977H.97c-1.295-.052-1.294-1.902 0-1.953h14.746c.54 0 .977.437.977.976Zm8.3-3.927c0 .85-.344 1.65-.97 2.254a3.254 3.254 0 0 1-1.568.842v.831c-.052 1.296-1.902 1.295-1.953 0v-.902a12.221 12.221 0 0 1-2.168-.631.977.977 0 0 1 .722-1.816c1.107.44 2.25.639 2.709.639.691 0 1.276-.557 1.276-1.216 0-.671-.688-1.217-1.532-1.217-1.01 0-1.893-.302-2.553-.875-2-1.703-.926-5.11 1.594-5.431v-.953c.052-1.296 1.902-1.295 1.954 0v1.069c.489.096 1.06.254 1.72.511 1.19.518.518 2.242-.707 1.82-1.2-.466-1.977-.526-2.52-.526-.644 0-1.13.528-1.13 1.228 0 .72.66 1.204 1.642 1.204 1.921 0 3.485 1.422 3.485 3.17Zm-12.5-16.483a1.22 1.22 0 0 0-1.22 1.221c.064 1.62 2.378 1.619 2.442 0a1.22 1.22 0 0 0-1.221-1.22Z"
            fill="currentColor"
          />
        </svg>
        <span
          class="appDrawer__menuText"
          style="opacity: 1; transform: none;"
        >
          Finance
        </span>
         
      </li>
      <li
        class=""
        role="menuitem"
      >
        <svg
          fill="none"
          height="26"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.47 16.174a1.22 1.22 0 0 1-1.22 1.22c-1.62-.064-1.619-2.377 0-2.44.674 0 1.22.546 1.22 1.22Zm2.98-1.22a1.22 1.22 0 0 0-1.221 1.22c.064 1.62 2.377 1.619 2.44 0a1.22 1.22 0 0 0-1.22-1.22Zm4.198 0a1.22 1.22 0 0 0-1.22 1.22c.064 1.62 2.377 1.619 2.441 0a1.22 1.22 0 0 0-1.22-1.22ZM6.25 19.151a1.22 1.22 0 0 0-1.22 1.221c.064 1.62 2.377 1.619 2.44 0a1.22 1.22 0 0 0-1.22-1.22Zm4.2 0a1.22 1.22 0 0 0-1.221 1.221c.064 1.62 2.377 1.619 2.44 0a1.22 1.22 0 0 0-1.22-1.22Zm4.198 0a1.22 1.22 0 0 0-1.22 1.221c.064 1.62 2.377 1.619 2.441 0a1.22 1.22 0 0 0-1.22-1.22Zm4.2 0a1.22 1.22 0 0 0-1.221 1.221c.064 1.62 2.377 1.619 2.441 0a1.22 1.22 0 0 0-1.22-1.22ZM25 11.632v8.009c0 3.23-2.628 5.859-5.86 5.859H5.86A5.866 5.866 0 0 1 0 19.64v-8.007c0-3.231 2.629-5.86 5.86-5.86h13.28c3.232 0 5.86 2.629 5.86 5.86Zm-1.953 0a3.91 3.91 0 0 0-3.906-3.905H5.859a3.91 3.91 0 0 0-3.906 3.906v8.008a3.91 3.91 0 0 0 3.906 3.906h13.282a3.91 3.91 0 0 0 3.906-3.906v-8.008Zm-2.93-.487H4.883c-1.296.051-1.295 1.902 0 1.953h15.234c1.296-.052 1.295-1.902 0-1.954ZM9.863 3.43a.977.977 0 1 1-1.953 0 .978.978 0 0 0-1.953 0 .977.977 0 1 1-1.953 0c.16-3.887 5.699-3.886 5.86 0ZM18.164.5a2.933 2.933 0 0 0-2.93 2.93.977.977 0 1 0 1.954 0 .978.978 0 0 1 1.953 0 .977.977 0 1 0 1.953 0A2.933 2.933 0 0 0 18.164.5Zm.684 14.453a1.22 1.22 0 0 0-1.221 1.22c.064 1.62 2.377 1.62 2.441 0a1.22 1.22 0 0 0-1.22-1.22Z"
            fill="currentColor"
          />
        </svg>
        <span
          class="appDrawer__menuText"
          style="opacity: 1; transform: none;"
        >
          Calendar
        </span>
         
      </li>
      <li
        class=""
        role="menuitem"
      >
        <svg
          fill="none"
          height="26"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.2 25.5c-9.48.046-14.704-11.358-8.525-17.901a2.71 2.71 0 0 1 2.583-.837 2.721 2.721 0 0 1 2.038 1.81c.03.089.042.164.05.234a.5.5 0 0 0 .66.185C9.564 8.177 9.767 6.298 9.97 3.3l.02-.285A2.662 2.662 0 0 1 11.508.772a2.68 2.68 0 0 1 2.72.217c3.82 2.65 5.631 6.628 6.37 8.763a.976.976 0 1 1-1.844.64c-.662-1.91-2.275-5.464-5.64-7.799a.743.743 0 0 0-.76-.061.725.725 0 0 0-.417.615l-.02.284c-.154 2.765-.424 6.004-3.006 7.29a2.455 2.455 0 0 1-2.048.106 2.428 2.428 0 0 1-1.4-1.466 1.057 1.057 0 0 1-.053-.253.78.78 0 0 0-.552-.435.769.769 0 0 0-.739.24c-6.274 7.002 2.057 17.795 10.942 13.793a.977.977 0 0 1 .815 1.774A11.12 11.12 0 0 1 11.2 25.5Zm10.274-3.84c2.256-1.689 3.471-3.711 3.514-5.85.026-2.8-3.386-4.245-5.37-2.314-1.986-1.932-5.395-.486-5.37 2.315.043 2.138 1.258 4.16 3.514 5.85.546.409 1.2.613 1.856.613.655 0 1.31-.205 1.856-.614Zm-4.053-7.097c.673 0 1.22.548 1.22 1.221.052 1.295 1.902 1.295 1.954 0 .067-1.616 2.365-1.62 2.44-.002-.043 1.908-1.509 3.4-2.731 4.315a1.15 1.15 0 0 1-1.372 0c-1.223-.916-2.688-2.407-2.732-4.315a1.225 1.225 0 0 1 1.221-1.219Zm5.126-10.36c.065-1.608 2.38-1.603 2.441.003-.023 1.166-.806 2.113-1.734 2.801-.418.31-.995.31-1.413 0-.928-.688-1.711-1.635-1.735-2.801.063-1.608 2.378-1.61 2.441-.004"
            fill="currentColor"
          />
        </svg>
        <span
          class="appDrawer__menuText"
          style="opacity: 1; transform: none;"
        >
          Timeline
        </span>
         
      </li>
      <li
        class=""
        role="menuitem"
      >
        <svg
          fill="none"
          height="26"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.14 25.5H5.86A5.866 5.866 0 0 1 0 19.64V6.36C0 3.128 2.629.5 5.86.5h13.28C22.373.5 25 3.129 25 6.36v13.28c0 3.232-2.628 5.86-5.86 5.86ZM5.86 2.453A3.91 3.91 0 0 0 1.952 6.36v13.282a3.91 3.91 0 0 0 3.906 3.906h13.282a3.91 3.91 0 0 0 3.906-3.906V6.359a3.91 3.91 0 0 0-3.906-3.906H5.859Zm.83 15.235a.977.977 0 0 1-.958-1.168l.553-2.767a7.04 7.04 0 0 1 1.931-3.61L12.91 5.45c1.327-1.382 3.714-1.382 5.041 0 1.383 1.327 1.383 3.715 0 5.041l-4.693 4.693a7.041 7.041 0 0 1-3.61 1.932l-2.766.553a.977.977 0 0 1-.192.018Zm8.74-11.329a1.6 1.6 0 0 0-1.14.472l-4.692 4.693A5.092 5.092 0 0 0 8.2 14.136l-.266 1.33 1.33-.266a5.092 5.092 0 0 0 2.612-1.397L16.57 9.11c.625-.6.625-1.679 0-2.279a1.6 1.6 0 0 0-1.14-.472Zm5.468 14.258a.977.977 0 0 0-.976-.976H5.078c-1.296.051-1.295 1.902 0 1.953h14.844c.54 0 .976-.438.976-.977Z"
            fill="currentColor"
          />
        </svg>
        <span
          class="appDrawer__menuText"
          style="opacity: 1; transform: none;"
        >
          Diary
        </span>
         
      </li>
      <li
        class=""
        role="menuitem"
      >
        <svg
          fill="none"
          height="26"
          width="25"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11.287 25.5a2.935 2.935 0 0 1-2.931-2.931v-.025a.839.839 0 0 0-.538-.784 9.405 9.405 0 0 1-2.478-1.443.845.845 0 0 0-.953-.085l-.006.003A2.944 2.944 0 0 1 .383 19.13a2.902 2.902 0 0 1-.275-2.224 2.91 2.91 0 0 1 1.385-1.769l.012-.007a.837.837 0 0 0 .42-.833 9.524 9.524 0 0 1 .008-2.688.848.848 0 0 0-.427-.85l-.016-.009A2.908 2.908 0 0 1 .108 8.982a2.903 2.903 0 0 1 .275-2.224A2.944 2.944 0 0 1 4.38 5.652l.059.033c.3.17.674.138.952-.084a9.41 9.41 0 0 1 2.422-1.4.842.842 0 0 0 .542-.785c.16-3.889 5.702-3.887 5.863 0v.02c0 .346.21.653.537.784a9.414 9.414 0 0 1 2.395 1.396.849.849 0 0 0 .954.086l.085-.048a2.945 2.945 0 0 1 4.002 1.104c.386.68.484 1.47.275 2.224a2.91 2.91 0 0 1-1.386 1.769l-.067.038a.833.833 0 0 0-.42.833 9.53 9.53 0 0 1 .01 2.648.839.839 0 0 0 .42.834l.059.033a2.91 2.91 0 0 1 1.384 1.768c.702 2.494-2.022 4.606-4.274 3.33l-.034-.019a.849.849 0 0 0-.958.089 9.412 9.412 0 0 1-2.449 1.437.84.84 0 0 0-.533.782v.044a2.935 2.935 0 0 1-2.932 2.932ZM4.8 18.166c.628 0 1.251.213 1.768.631.595.481 1.256.866 1.964 1.144a2.782 2.782 0 0 1 1.778 2.603v.024c.054 1.297 1.9 1.296 1.954 0v-.044c0-1.15.693-2.17 1.765-2.597a7.457 7.457 0 0 0 1.94-1.14 2.79 2.79 0 0 1 3.155-.27l.031.017c.47.265 1.07.1 1.337-.37a.968.968 0 0 0-.373-1.326l-.06-.035a2.773 2.773 0 0 1-1.39-2.798 7.575 7.575 0 0 0-.009-2.103A2.8 2.8 0 0 1 20.05 9.09l.07-.04a.97.97 0 0 0 .373-1.327.986.986 0 0 0-1.34-.368l-.082.047c-1 .568-2.233.466-3.14-.26a7.459 7.459 0 0 0-1.899-1.107 2.787 2.787 0 0 1-1.767-2.599v-.02c-.053-1.296-1.9-1.296-1.954 0A2.785 2.785 0 0 1 8.53 6.02a7.46 7.46 0 0 0-1.92 1.11 2.794 2.794 0 0 1-3.135.255l-.056-.032a.985.985 0 0 0-1.337.37.969.969 0 0 0 .37 1.325l.02.01c.999.568 1.56 1.706 1.395 2.833a7.567 7.567 0 0 0-.007 2.134 2.77 2.77 0 0 1-1.39 2.804l-.015.008a.971.971 0 0 0-.373 1.327.986.986 0 0 0 1.34.369 2.786 2.786 0 0 1 1.378-.367Zm6.487-.044a5.136 5.136 0 0 1-5.13-5.13c.282-6.805 9.98-6.803 10.26 0a5.136 5.136 0 0 1-5.13 5.13Zm0-8.305a3.18 3.18 0 0 0-3.176 3.175c.175 4.213 6.178 4.212 6.352 0a3.18 3.18 0 0 0-3.176-3.175Z"
            fill="currentColor"
          />
        </svg>
        <span
          class="appDrawer__menuText"
          style="opacity: 1; transform: none;"
        >
          Settings
        </span>
         
      </li>
    </ul>
  </div>
  <div
    class="flex flex-col"
  >
    <label
      class="appDrawer__toggler swap swap-rotate"
    >
      <input
        checked=""
        data-testid="drawer-toggler"
        type="checkbox"
      />
      <svg
        class="swap-off fill-current"
        height="25"
        viewBox="0 0 512 512"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M64,384H448V341.33H64Zm0-106.67H448V234.67H64ZM64,128v42.67H448V128Z"
        />
      </svg>
      <svg
        class="swap-on fill-current"
        height="25"
        viewBox="0 0 512 512"
        width="25"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="400 145.49 366.51 112 256 222.51 145.49 112 112 145.49 222.51 256 112 366.51 145.49 400 256 289.49 366.51 400 400 366.51 289.49 256 400 145.49"
        />
      </svg>
    </label>
  </div>
</div>
`;
